README.Rmd

```diff
@@ -13,108 +13,74 @@ knitr::opts_chunk$set(
-[![](https://www.r-pkg.org/badges/version/keyring)](https://www.r-pkg.org/pkg/keyring)
-[![CRAN RStudio mirror downloads](https://cranlogs.r-pkg.org/badges/keyring)](https://www.r-pkg.org/pkg/keyring)
-[![Codecov test coverage](https://codecov.io/gh/r-lib/keyring/branch/main/graph/badge.svg)](https://app.codecov.io/gh/r-lib/keyring?branch=main)
-<!-- badges: end -->
-
-Platform independent API to access the operating systems
-credential store. Currently supports:
-
-* Keychain on macOS (`backend_macos`),
-* Credential Store on Windows (`backend_wincred`),
-* the Secret Service API on Linux (`backend_secret_service`),
-* encrypted files (`backend_file`), and
-* environment variables (`backend_env`).
 
-The last two are available on all platforms.
-Additional storage backends can be added easily.
+[![R-CMD-check](https://github.com/r-lib/keyring/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/r-lib/keyring/actions/workflows/R-CMD-check.yaml) [![](https://www.r-pkg.org/badges/version/keyring)](https://www.r-pkg.org/pkg/keyring) [![CRAN RStudio mirror downloads](https://cranlogs.r-pkg.org/badges/keyring)](https://www.r-pkg.org/pkg/keyring) [![Codecov test coverage](https://codecov.io/gh/r-lib/keyring/branch/main/graph/badge.svg)](https://app.codecov.io/gh/r-lib/keyring?branch=main)
 
-## Installation
-
-### Linux
-
-For the Secret Service backend install the `libsecret` library, at least
-version 0.16. This is typically the best keyring for a Linux desktop.
-Install these packages:
+<!-- badges: end -->
 
-- Debian/Ubuntu: `libsecret-1-dev`
-- Recent RedHat, Fedora and CentOS systems: `libsecret-devel`
+keyring provides a way to securely manage secrets using your operating system's credential store.
+Once a secret is defined, it persists in the keyring across multiple R sessions.
+keyring is an alternative to using env vars that's a bit more secure because your secret is never stored in plain text, meaning that you can (e.g.) never accidentally upload it to GitHub.
 
-The file backend uses the sodium package, which needs the sodium library.
-This backend works best on Linux servers. Install these packages:
+keyring currently supports:
 
-- Debian/Ubuntu: `libsodium-dev`
-- Fedora, EPEL: `libsodium-devel`
+-   The macOS Keychain (`backend_macos`).
+-   The Windows Credential Store (`backend_wincred`).
+-   The Linux Secret Service API (`backend_secret_service`).
 
-### OS X and Windows
+It also provides two backends that are available on all platforms:
 
-No additional software is needed.
+-   Encrypted files (`backend_file`)
+-   Environment variables (`backend_env`).
 
-### R package
+## Installation
 
 Install the package from CRAN:
 
 ```{r eval = FALSE}
-install.packages("keyring")
+# install.packages("pak")
+pak::pak("keyring")
 ```
 
-## Usage
+We recommend using pak to install keyring as it will ensure that all Linux system requirements (e.g. `libsecret-devel`, `openssl-devel`) are automatically installed.
```

Comment:
If we are mentioning them already, we can maybe mention `libsodium-devel` as well, and maybe also note that these are for (say) Fedora 38.
```
❯ pak::pkg_sysreqs("keyring", sysreqs_platform = "fedora-38")
── Install scripts ──────────────────────────────────────────────── Fedora 38 ──
dnf install -y libsecret-devel openssl-devel libsodium-devel

── Packages and their system dependencies ──────────────────────────────────────
keyring – libsecret-devel
openssl – openssl-devel
sodium  – libsodium-devel
```

```
❯ pak::pkg_sysreqs("keyring", sysreqs_platform = "ubuntu-2204")
── Install scripts ────────────────────────────────────────────── Ubuntu 2204 ──
apt-get -y update
apt-get -y install libsecret-1-dev libssl-dev libsodium-dev

── Packages and their system dependencies ──────────────────────────────────────
keyring – libsecret-1-dev
openssl – libssl-dev
sodium  – libsodium-dev
```